optimizers/optimizer.py

```diff
@@ -61,7 +61,7 @@ def __init__(
   ):
     self.optimizer = optimizer
     self.lr_dict = lr_dict
-    self.group_names = list(self.lr_dict.keys())
+    self.group_names = [* self.lr_dict]
```

Comment:
This way compromises the readability

Reply:
Yeah, there is a tradeoff between readability and performance. It is up to the team to decide which to maximize. You may argue that I would prefer readability for maintainability over that little performance gain, and that's totally fine.

Reply:
There's no practical performance benefit from doing it this way.
It's trivial to benchmark the difference:
```python
In [4]: def test1():
   ...:     return list(a.keys())
   ...: 

In [5]: def test2():
   ...:     return [*a]
   ...: 

In [6]: %timeit -n 10_000_000 test2
23.1 ns ± 0.214 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [7]: %timeit -n 10_000_000 test1
23.1 ns ± 0.246 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

using ipython

Reply:
```sh
>>> from timeit import timeit
>>> timeit("[]")
0.09845466999831842
>>> timeit("list()")
0.22419986899876676
```

Reply:
That is simply not the same code at all, you're comparing apples to oranges.
My benchmark shows the exact change you did, and shows no performance improvement in the slightest.

In fact, if you run the benchmark using Python 3.11 the older method is consistently faster.

```python
Python 3.11.2 (main, Feb  8 2023, 14:49:25) [GCC 11.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.12.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: a = {"a": 5, "k": 3, "asdf": 53}

In [2]: def test1():
   ...:     return list(a.keys())
   ...: 

In [3]: def test2():
   ...:     return [*a]
   ...: 

In [4]: %timeit -n 10_000_000 test2
13.7 ns ± 1.16 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [5]: %timeit -n 10_000_000 test1
13 ns ± 1.15 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [6]: %timeit -n 10_000_000 test1
13 ns ± 1.22 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [7]: %timeit -n 10_000_000 test2
14.1 ns ± 1.15 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

```
In [12]: dis.dis(test1)
  1           0 RESUME                   0

  2           2 LOAD_GLOBAL              1 (NULL + list)
             14 LOAD_GLOBAL              2 (a)
             26 LOAD_METHOD              2 (keys)
             48 PRECALL                  0
             52 CALL                     0
             62 PRECALL                  1
             66 CALL                     1
             76 RETURN_VALUE

In [13]: dis.dis(test2)
  1           0 RESUME                   0

  2           2 BUILD_LIST               0
              4 LOAD_GLOBAL              0 (a)
             16 LIST_EXTEND              1
             18 RETURN_VALUE
```

While test2 has a smaller bytecode output, it is still slower than test1. Not all opcodes are equally fast.

Reply:
It seems like python 3.11 introduced a lot of optimizations out of the box, and it seems like it is the default behavior under the hood. On python 3.9, I am getting these results:

```sh
>>> timeit('[* {"a": 5, "k": 3, "asdf": 53}]')
0.7558078520014533
>>> timeit('list({"a": 5, "k": 3, "asdf": 53})')
0.934761193999293
```

Reply:
Even if the change were to have a minor performance benefit, none of the code paths seem to be hot, so these nano optimizations gain nothing in practice.